src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1129,4 +1138,19 @@ def check_build_log_access
 
     true
   end
+
+  def check_repository_and_architecture
```

Comment:
Out of scope for this PR, but I've seen this kind of validation so many times. It might be worth to find a generic way to handle this.